packages/mattermost-redux/src/selectors/entities/general.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {General} from 'mattermost-redux/constants';
+
+import {GlobalState} from 'mattermost-redux/types/store';
+import {ClientConfig, FeatureFlags} from 'mattermost-redux/types/config';
+
+import {isMinimumServerVersion} from 'mattermost-redux/utils/helpers';
+
+export function getConfig(state: GlobalState): Partial<ClientConfig> {
```

Comment:
There seems to be some argument around types for `state.entities.general.config` and `getConfig` since this function previously was defined as returning a full `ClientConfig` but the redux state held a partial one. The partial seems correct because the client config might not be loaded or might be missing fields when not logged in, but I wasn't entirely sure about that.

It might be something to consider changing in the future when we properly type the values coming from the server instead of just sending down a load of strings.